pkg/library/libraryinputresources/prune_mustgather.go

```diff
@@ -52,6 +53,40 @@ func GetRequiredInputResourcesFromMustGather(ctx context.Context, inputResources
 		return nil, err
 	}
 
+	inputDirResources, err := LenientResourcesFromDirRecursive(mustGatherDir)
+	if err != nil {
+		return nil, err
+	}
+
+	var flattenedInputDirResources []*Resource
+	for _, inputDirResource := range inputDirResources {
+		if !inputDirResource.Content.IsList() {
+			flattenedInputDirResources = append(flattenedInputDirResources, inputDirResource)
+			continue
+		}
+		list, err := inputDirResource.Content.ToList()
+		if err != nil {
+			return nil, err
+		}
+		for _, item := range list.Items {
+			singleResource := &Resource{
+				Filename:     inputDirResource.Filename,
+				ResourceType: inputDirResource.ResourceType,
+				Content:      &item,
+			}
+			flattenedInputDirResources = append(flattenedInputDirResources, singleResource)
+		}
+	}
+
+	differences := DifferenceOfResources(flattenedInputDirResources, pertinentUnstructureds)
```

Comment:
I think we want this check only when we're running tests and as a separate check.

Actually we probably want it on general apply-configuration commands, but it should still be a clear and separate call to get the list of extra inputs